recipes/kerberos_init.rb

```diff
@@ -27,8 +27,8 @@
   Chef::Log.info("Secure Hadoop Enabled: Kerberos Realm '#{node['krb5']['krb5_conf']['realms']['default_realm']}'")
   secure_hadoop_enabled = true
 
-  # Create users for services
-  %w(hadoop hbase hdfs hive mapred spark yarn zookeeper).each do |u|
+  # Create users for services not in hadoop::default
```

Comment:
The problem wasn't the user, it was the group. If the group exists, but the user doesn't, it causes a failure, because you need a different invocation of useradd, which Chef doesn't do.